src/coordinator/ts/values.go

```diff
@@ -39,79 +38,98 @@ type Values interface {
 	// ValueAt returns the value at the nth element
 	ValueAt(n int) float64
 
-	// The number of millisseconds represented by each index
-	MillisPerStep() int
-
-	// Slice of data values in a range
-	Slice(begin, end int) Values
+	// DatapointAt returns the datapoint at the nth element
+	DatapointAt(n int) Datapoint
+}
 
-	// AllNaN returns true if the values are all NaN
-	AllNaN() bool
+// A Datapoint is a single data value reported at a given time
+type Datapoint struct {
+	Timestamp time.Time
+	Value     float64
 }
 
+// Datapoints is a list of datapoints.
+type Datapoints []Datapoint
+
+// Len is the length of the array.
+func (d Datapoints) Len() int { return len(d) }
+
+// ValueAt returns the value at the nth element.
+func (d Datapoints) ValueAt(n int) float64 { return d[n].Value }
+
+// DatapointAt returns the value at the nth element.
+func (d Datapoints) DatapointAt(n int) Datapoint { return d[n] }
+
 // MutableValues is the interface for values that can be updated
 type MutableValues interface {
 	Values
 
-	// Resets the values
-	Reset()
-
 	// Sets the value at the given entry
 	SetValueAt(n int, v float64)
 }
 
-// NewValues returns MutableValues supporting the given number of values at the
-// requested granularity.  The values start off as NaN
-func NewValues(ctx context.Context, millisPerStep, numSteps int) MutableValues {
-	return newValues(ctx, millisPerStep, numSteps, math.NaN())
+// FixedResolutionMutableValues are mutable values with fixed resolution between steps
+type FixedResolutionMutableValues interface {
```

Comment:
Can this be a struct rather than an interface if nothing else creates these?

Reply:
just taking inspiration from statsdex, we are going to have more things implement it. Eg: constant values,